File: tests/fixtures/python/policies/pii_redaction_before_llm/compliant_decorated/app/handlers.py
"""Compliant via decorator-based guard: @redacted on the function."""
import anthropic


def redacted(fn):
    return fn


@redacted
def summarize_user(user_id: str) -> str:
    user = load_user(user_id)
    return anthropic.messages.create(model="claude", input=user)


def load_user(user_id: str) -> dict:
    return {"id": user_id, "email": "x@example.com"}

File: tests/fixtures/typescript/policies/pii_redaction_before_llm/violating_arrow/src/handlers.ts
import * as anthropic from 'anthropic';

export const summarizeUser = (userId: string): string => {
  const user = loadUser(userId);
  return anthropic.messages.create({ model: 'claude', input: user });
};

export const loadUser = (userId: string) => ({ id: userId, email: 'x@example.com' });
